extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeSuggestionExtractor.java

```diff
@@ -45,17 +45,21 @@ public YoutubeSuggestionExtractor(StreamingService service) {
 
     @Override
     public List<String> suggestionList(String query) throws IOException, ExtractionException {
-        Downloader dl = NewPipe.getDownloader();
-        List<String> suggestions = new ArrayList<>();
+        final Downloader dl = NewPipe.getDownloader();
+        final List<String> suggestions = new ArrayList<>();
 
-        String url = "https://suggestqueries.google.com/complete/search"
+        final String url = "https://suggestqueries.google.com/complete/search"
                 + "?client=" + "youtube" //"firefox" for JSON, 'toolbar' for xml
                 + "&jsonp=" + "JP"
                 + "&ds=" + "yt"
                 + "&gl=" + URLEncoder.encode(getExtractorContentCountry().getCountryCode(), UTF_8)
                 + "&q=" + URLEncoder.encode(query, UTF_8);
 
-        String response = dl.get(url, getExtractorLocalization()).responseBody();
+        // CONSENT header to prevent redirect to consent.youtube.com
+        final Map<String, List<String>> headers = new HashMap<>();
+        headers.put("CONSENT", Collections.singletonList(CONSENT_COOKIE_VALUE));
```

Comment:
Maybe add a dedicated function to YoutubeParsingHelper?

Reply:
Done